packages/serverpod/lib/src/database/database_connection.dart

```diff
@@ -629,13 +630,20 @@ class DatabaseConnection {
     if (column is ColumnString) return 'text';
     if (column is ColumnBool) return 'boolean';
     if (column is ColumnInt) return 'integer';
-    if (column is ColumnEnum) return 'integer';
     if (column is ColumnDouble) return 'double precision';
     if (column is ColumnDateTime) return 'timestamp without time zone';
     if (column is ColumnByteData) return 'bytea';
     if (column is ColumnDuration) return 'bigint';
     if (column is ColumnUuid) return 'uuid';
     if (column is ColumnSerializable) return 'json';
```

Comment:
Does `ColumnEnumExtended` extend `ColumnSerializable`? If so, the `ColumnSerializable` check needs to come last.